lm_eval/caching/cache.py

```diff
@@ -0,0 +1,46 @@
+import os
+import dill
+
+
+MODULE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+OVERRIDE_PATH = os.getenv("LM_HARNESS_CACHE_PATH")
+
+
+PATH = OVERRIDE_PATH if OVERRIDE_PATH else f"{MODULE_DIR}/.cache"
+
+FILE_SUFFIX = ".pickle"
+
+
+def load_from_cache(file_name):
+    try:
+        path = f"{PATH}/{file_name}.pickle"
+
+        with open(path, "rb") as file:
+            cached_task_dict = dill.loads(file.read())
+            return cached_task_dict
+
+    except Exception as exception:
+        print(f"{file_name} is not cached, generating...")
+        pass
+
+
+def save_to_cache(file_name, obj):
+    if not os.path.exists(PATH):
+        os.mkdir(PATH)
+
+    file_path = f"{PATH}/{file_name}{FILE_SUFFIX}"
+
+    print(f"Saving {file_path} to cache...")
+    with open(file_path, "wb") as file:
+        file.write(dill.dumps(obj))
+
+
+# NOTE the "key" param is to allow for flexibility
+def delete_cache(key: str = ""):
```

Comment:
I wonder whether `delete_cache` might be better to 
1. only delete the requests for the tasks you are running with
2. need to take a key, and as key / filename use the base64 encoding or something of the taskname? would want to minimize any risk of delete_cache() deleting user files, although I think this is already pretty good with the default location being caching/.cache

Reply:
I'd be glad to add this in a subsequent batch of work, but I think the current speed up is already a big improvement. In the meantime, users can just manually delete the files they want to refresh in their code editor

#2 is now taken care of!